Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -403,27 +403,71 @@ public final class AppDelegateViewModel: AppDelegateViewModelType, AppDelegateVi
     self.findRedirectUrl = deepLinkUrl
       .filter { Navigation.match($0) == .emailClick }
 
+    /// INFO: Handles the deeplink route with both an id and text based name for a deeplink for categories.
     let deepLinkCategories: ([String: String]) -> (Param?, Param?) = { rawParams in
       let parentCategoryParams = rawParams["parent_category_id"]
       let subCategoryParams = rawParams["category_id"]
       var categoryParam: Param?
       var subcategoryParam: Param?
 
-      switch (parentCategoryParams, subCategoryParams) {
-      case let (.some(rawCategoryParams), .some(rawSubcategoryParams)):
-        categoryParam = .some(Param.slug(rawCategoryParams))
-        subcategoryParam = .some(Param.slug(rawSubcategoryParams))
-      case (let .some(rawCategoryParams), nil):
-        categoryParam = .some(Param.slug(rawCategoryParams))
-      case (nil, let .some(rawSubcategoryParams)):
-        categoryParam = .some(Param.slug(rawSubcategoryParams))
-      case (nil, nil):
-        return (nil, nil)
+      let rawId: (String?) -> Int? = { rawParam in
```

Comment:
This is quite a bit of logic for an inline closure. I still think this and the method below should be an explicit method declaration.